hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/utils/AutoCompletion.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.utils;
+
+import java.util.Objects;
+import org.apache.hadoop.hdds.cli.GenericCli;
+import org.apache.ratis.util.ReflectionUtils;
+import org.reflections.Reflections;
+import picocli.AutoComplete;
+import picocli.CommandLine;
+import picocli.CommandLine.Option;
+
+/**
+ * Tool to generate bash/zsh auto-complete scripts for Ozone CLI.
+ */
+public final class AutoCompletion {
+
+  private static final String OZONE_COMMAND = "ozone ";
+  private static final int PREFIX_LENGTH = OZONE_COMMAND.length();
+  private static final String[] PACKAGES_TO_SCAN = {
+      "org.apache.hadoop.hdds",
+      "org.apache.hadoop.ozone",
+      "org.apache.ozone",
+  };
+
+  private AutoCompletion() { }
+
+  public static void main(String[] args) {
+    final CommandLine ozone = new CommandLine(new Ozone());
+    for (String pkg : PACKAGES_TO_SCAN) {
+      new Reflections(pkg).getSubTypesOf(GenericCli.class).stream()
+          .map(AutoCompletion::getCommand)
+          .filter(Objects::nonNull)
+          .filter(AutoCompletion::isPublicCommand)
+          .forEach(command -> ozone.addSubcommand(getCommandName(command), command));
+    }
+    System.out.println(AutoComplete.bash("ozone", ozone));
```

Comment:
Would it be better to write to a file directly?  Running `ozone completion` outputs 364KB of content to the console.

```java
    File file = new File("ozone-completion.sh");
    AutoComplete.bash("ozone", file, null, ozone);
```

Reply:
The reason for writing the script to stdout is to allows users to install Ozone completion with a single command. The same command can also be added to the `.bashrc`/`.zshrc` file.
```
source <(ozone completion)
```